lib/modules/manager/swift/util.ts

```diff
@@ -0,0 +1,33 @@
+import { regEx } from '../../../util/regex';
+
+const SWIFT_TOOLS_VERSION_REGEX = regEx(
+  /^\s*\/\/\s*swift-tools-version:\s*(?<version>\d+\.\d+(?:\.\d+)?)/
+);
+
+const SWIFT_TOOLS_SEMVER_REGEX = regEx(/\d+\.\d+\.\d+/);
+
+/**
+ * Extracts the Swift tools version from the given package file contents as a semver version.
+ *
+ * Tools version specification:
+ * https://github.com/apple/swift-package-manager/blob/main/Documentation/Usage.md#swift-tools-version-specification
+ */
+export function extractSwiftToolsVersion(
+  packageFileContent: string
+): string | null {
+  const toolsVersionMatch = SWIFT_TOOLS_VERSION_REGEX.exec(packageFileContent);
+
+  if (toolsVersionMatch?.groups) {
+    const version = toolsVersionMatch.groups.version;
+
+    // The version is either '<major>.<minor>.<patch>' or '<major>.<minor>'
+    // Ensure we always return a semver compatible version.
+    if (version.match(SWIFT_TOOLS_SEMVER_REGEX)) {
+      return version;
+    } else {
+      return version + '.0';
```

Comment:
```suggestion
      return `^${version}.0`;
```
i think we should pass a range to allow latest patch

Reply:
Using a range works but will require using npm versioning for the swift tool docker datasource instead of semver since semver doesn't support ranges.
I guess this should still work since npm versioning will see the docker "-*" suffixed tags as prerelease versions and will still reject them.

Separately, I would argue for a `>` range instead of `^`. For now, swift package manager is backwards compatible across major versions. This might need to be changed if that ever changes in the future.

In the meantime, it seems more likely to me that renovate might fail to perform lockfile updates if an updated dependency requires a new major tool version and we limit ourselves to the major version used by the local package.

Artificial example:
* Local package tools version (this is the version specified in the package being renovated and is the minimum tools version needed to use the package): 5.7.2
* Newest global tools version: 6.x
* Assume the local package uses a dependency that just got updated and the newest version of that dependency now requires a tool version >=6
* Now renovate would fail to create the new lockfile for the dependency update PR since the tool version we would use (^5.x) would error on a dependency requiring a >=6 tools version.

Technically the local package file in that case is incorrect in that the minimum tools version specified is not actually correct once the (transitive) dependencies are taken into account but nothing prevents that from happening and is easy to overlook as long as everyone is using a new enough tools version that works with all dependencies.

Long term renovate might help here by also managing the tools version in the package file.

Reply:
I added the range constraint using `>=` as described above.
Let me know if that seems sensible to you and if I should add some comments to document the reasoning behind it or if you'd prefer that I change the constraint to `^`.

Reply:
ping @viceice